pkg/controllers/nodeclaim/disruption/drift_test.go

```diff
@@ -499,5 +505,12 @@ var _ = Describe("Drift", func() {
 			nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
 			Expect(nodeClaim.StatusConditions().GetCondition(v1beta1.Drifted)).To(BeNil())
 		})
+		It("should not return drifted if karpenter.sh/nodePool-hash annotation version does not match the nodeClaim", func() {
```

Comment:
Do we need a test to validate that we stay drifted in the scenario where we have already drifted and we get a change in the hash annotation version?